src/components/skills-quiz/tests/SkillsQuiz.test.jsx

```diff
@@ -86,4 +87,20 @@ describe('<SkillsQuiz />', () => {
     );
     expect(screen.getByText(SKILLS_QUIZ_SEARCH_PAGE_MESSAGE)).toBeTruthy();
   });
+
+  it('renders skills quiz V2 page successfully for v2', () => {
+    // Set the feature flag to enable v2
+    // eslint-disable-next-line global-require
+    jest.spyOn(require('@edx/frontend-enterprise-utils'), 'hasFeatureFlagEnabled').mockImplementation(() => true);
```

Comment:
When mocking, typically the import (`require(...)`) shouldn't be done in the test itself. Can you setup the mock for `hasFeatureFlagEnabled` in a more standardized way?

```jsx
import { hasFeatureFlagEnabled } from '@edx/frontend-enterprise-utils';

// this mock setup already exists in the file...
jest.mock('@edx/frontend-enterprise-utils', () => ({
  ...jest.requireActual('@edx/frontend-enterprise-utils'),
  sendEnterpriseTrackEvent: jest.fn(),
  // can simply add `hasFeatureFlagEnabled` here
  hasFeatureFlagEnabled: jest.fn().mockReturnValue(false),
}));

it('renders skills quiz V2 page...', () => {
  hasFeatureFlagEnabled.mockReturnValue(true);
  // ...
});
```

Also worth noting, it's not quite ideal to only rely on `hasFeatureFlagEnabled` to determine whether a feature flag is enabled; in the tests, if your component/feature were using multiple feature flags, only controlled by `hasFeatureFlagEnabled`, your tests would be enabling _both_ features together, not each individual feature flag (a bit brittle). Generally, we should start relying more on Waffle-based feature flags so that the flags can toggled on/off in a more granular fashion without the need for query-parameter based enablement (no need to change for the current feature flags, though). See https://github.com/openedx/edx-enterprise/blob/master/docs/decisions/0012-enterprise-feature-flags-waffle.rst for more details.

Reply:
@adamstankiewicz Thanks for the input! We opted for featureFlag to run both quiz versions concurrently. 
Please note that we are already working on a more granular approach, an `Optimizely` experiment for a 50/50 A/B test is in progress to run different versions of skills quiz page.
Here are the PRs:
- https://github.com/edx/edx-internal/pull/10069
- https://github.com/openedx/frontend-app-learner-portal-enterprise/pull/920